### ISSUE TITLE
Notification banner in cli that informs of later versions when released

### ISSUE THREAD

**Issue by practiv-jtew** [Branchout/branchout#60]
Branchout should include a notification if there are more recent versions

```
> branchout init
There is an update to branchout, version 5.1 is available. 

etc/etc/...

```

**Comment by stickycode** [comment]
need to call this 

curl --silent "https://api.github.com/repos/Branchout/branchout/releases/latest"

and grab the tag name, probably want to do it once an hour and store the result in BRANCHOUT_STATE/version-check

**Comment by fredcooke** [comment]
Agreed on caching it if doing it at all. Daily, though - This tends to only get released multiple times in a day when something gets screwed up so you'll get it tomorrow either way if daily. Could even be backgrounded and store the cached result not necessarily showing it on the first run of the day.

This should 100% be accompanied by a way to turn it off so ET doesn't phone home for sensitive environments and sensitive people and people who don't want a sporadic performance hit once an hour or once a day (more appropriate) and for people who are capable of managing their own update timings without software nagging.
